project1/ConstantBSProcess.cpp

```diff
@@ -0,0 +1,66 @@
+#include <blackscholesprocess.hpp>
+#include <ConstantBSProcess.h>
+
+ConstantBSProcess::ConstantBSProcess(Real S,Rate r,Volatility v,Rate q){
+S = 100.0;
+r = process_->riskFreeRate()->zeroRate(grid.back(), Continuous);
+v = process_->blackVolatility()->blackVol(grid.back(), Continuous);
+q = process_->dividendYield()->zeroRate(grid.back(), Continuous);
+}
```

Comment:
The idiomatic way to assign the passed parameters to the data members in C++ is:
```suggestion
ConstantBSProcess::ConstantBSProcess(Real S,Rate r,Volatility v,Rate q)
: S(S), r(r), v(v), q(q) {}
```
You don't have a process here; you'll extract the rates and volatility in the engine before passing them.